src/devices/src/virtio/block/device.rs

```diff
@@ -352,12 +352,13 @@ impl VirtioDevice for Block {
     fn write_config(&mut self, offset: u64, data: &[u8]) {
         let data_len = data.len() as u64;
         let config_len = self.config_space.len() as u64;
-        if offset + data_len > config_len {
+        if offset + data_len != config_len {
```

Comment:
This doesn't seem right, now we either write the whole config or we don't.

Reply:
I'm for refusing to write partial configs. Since Firecracker only supports 1 parameter in each config space (capacity for blk; mac addr for net) I think we should enforce writing the full config space - 8 bytes or 6, respectively. Per a brief offline discussion with @sandreim we could further restrict (and simplify) this check by accepting only write requests from offset 0 and of length equal to the config space size. @firecracker-microvm/compute-capsule WDYT?